http/file_server.ts

```diff
@@ -233,16 +251,20 @@ function fileLenToString(len: number): string {
  * Returns an HTTP Response with the requested file as the body.
  * @param req The server request context used to cleanup the file handle.
  * @param filePath Path of the file to serve.
+ * @param etagAlgorithm The algorithm to use for generating the ETag. Defaults to "fnv1a".
+ * @param fileInfo An optional FileInfo object returned by Deno.stat. It is used
+ * for optimization purposes.
  */
 export async function serveFile(
   req: Request,
   filePath: string,
+  etagAlgorithm: EtagAlgorithm = "fnv1a",
+  fileInfo?: Deno.FileInfo,
```

Comment:
These two should be an option bag according to our style guide I think (public api)